src/hotspot/share/c1/c1_LIRGenerator.cpp

```diff
@@ -3043,31 +3047,131 @@ void LIRGenerator::do_IfOp(IfOp* x) {
 
 #ifdef JFR_HAVE_INTRINSICS
 void LIRGenerator::do_ClassIDIntrinsic(Intrinsic* x) {
```

Comment:
Is this method really important and hot enough that we need to intrinsify it both in C1 and C2?

Reply:
This optimization works when user-defined events contain fields of class type, so I think it is necessary to implement it in both C1 and C2